blocks/product-carousel/product-carousel.js

```diff
@@ -0,0 +1,227 @@
+import {
+  addEnclosingDiv,
+  createSlideSlider,
+  createSliderWrapper,
+  initSlider,
+  getSliderChildren,
+  setSlideDuration,
+  createArrowNav,
+  setLeftAndRightArrowHtml,
+  setSliderChildren,
+} from '../../scripts/lib-carousel.js';
+import { createOptimizedPicture } from '../../scripts/lib-franklin.js';
+
+// Large sreen bigger than px
+const LARGE_SCREEN = 1000;
+
+// Number of required column and rows in table
+const NUM_COLUMNS = 2;
+const NUM_ROWS = 2;
+
+// Error Messages
+const INVALID_NUMBER_OF_COLUMNS_MESSAGE = `Invalid configuration. Table with ${NUM_COLUMNS} columns and at least 1 row required`;
+const INVALID_NUMBER_OF_ROWS_MESSAGE = `Invalid configuration. At least ${NUM_ROWS} rows required to properly show the Product Carousel`;
+
+// HTML for arrow navigation
+const HTML_LEFT_ARROW = '<svg fill="rgb(88,127,194)" width="24px" height="24px" viewBox="0 0 1024 1024" class="icon"  xmlns="http://www.w3.org/2000/svg">'
+  + '<path d="M768 903.232l-50.432 56.768L256 512l461.568-448 50.432 56.768L364.928 512z"/>'
+  + '</svg>';
+
+const HTML_RIGHT_ARROW = '<svg fill="rgb(88,127,194)" width="24px" height="24px" viewBox="0 0 1024 1024" class="icon"  xmlns="http://www.w3.org/2000/svg">'
+  + '<path d="M256 120.768L306.432 64 768 512l-461.568 448L256 903.232 659.072 512z"/>'
+  + '</svg>';
+
+/**
+ * Get optimized img element width default
+ * @param picture
+ */
+export function optimizeThumbnails(picture) {
+  picture
+    .querySelectorAll('img')
+    .forEach((img) => {
+      const imgHeight = Math.floor((img.height * 1024) / img.width);
+      img
+        .closest('picture')
+        .replaceWith(
+          createOptimizedPicture(
+            img.src,
+            'Slider Image',
+            true,
+            img.width,
+            imgHeight,
+          ),
+        );
+    });
+}
+
+/**
+ * Check if the block has the correct number of columns and rows
+ * @param block
+ */
+function checkConfig(block) {
+  if (block.children.length === 0 || block.children[0].children.length < NUM_COLUMNS) {
+    throw new Error(INVALID_NUMBER_OF_COLUMNS_MESSAGE);
+  }
+  if (block.children.length < NUM_ROWS) {
+    throw new Error(INVALID_NUMBER_OF_ROWS_MESSAGE);
+  }
+}
+
+/**
+ * Move Elements in an array by a given number of positions.
+ * Positive numPositions value moves elements to the right,
+ * negative numPositions value moves elements to the left.
+ * @param arr
+ * @param numPositions
+ * @returns {*}
+ */
+function moveArrayElements(arr, numPositions) {
```

Comment:
this can be simplified by using the spread operator and `slice`:

```
function moveArrayElements(arr, numPositions) {
  const length = arr.length;
  if (length < 2) return arr;

  const normalizedPositions = numPositions % length;
  return normalizedPositions > 0
    ? [...arr.slice(-normalizedPositions), ...arr.slice(0, length - normalizedPositions)]
    : [...arr.slice(-normalizedPositions), ...arr.slice(0, -normalizedPositions)];
}
```

Reply:
your suggestion breaks navigation in mobile view. I changed it to use `slice`:

```
function moveArrayElements(arr, numPositions) {
  const length = arr.length;
  if (length < 2) {
    return arr;
  }

  const normalizedPositions = numPositions % length;
  if (normalizedPositions === 0) {
    return arr;
  }

  if (normalizedPositions > 0) {
    const movedElements = arr.splice(length - normalizedPositions);
    arr.unshift(...movedElements);
  } else {
    const movedElements = arr.splice(0, -normalizedPositions);
    arr.push(...movedElements);
  }

  return arr;
}````